Cleaned contract and testing, updated readme, decouple verify

Raihan9797 opened the pull request:
Remove unecessary commented out code from contract and test file
Update readme to install and run our code
decouple verify() into a check_user_list(), and verify().

Raihan9797 committed:
update readme for installation

Raihan9797 committed:
create habit on front end of type 0

Raihan9797 committed:
cleaning test file

Raihan9797 committed:
decoupled verify into tick_user_list and verify. Also imple get all losers pool

ambhinav commented on test/test_habit.js:
```diff
@@ -135,55 +140,34 @@ contract('Habit', function(accounts) {
         // loop through array of big numbers
-        for (let i = 0; i < is_user1_array2.length; i++) {
-            console.log(is_user1_array2[i].toNumber());
+        console.log('user 1 checklist')
```
change to event check

ambhinav commented on test/test_habit.js:
```diff
@@ -135,55 +140,34 @@ contract('Habit', function(accounts) {
+        let user2_day2_verified = await habit_instance.tick_user_list(0, accounts[2], 2);
 
+        let is_user2_array = await habit_instance.get_user_check_list.call(0, accounts[2]);
+        console.log('user 2 checklist')
```
change to event check

Raihan9797 committed:
emit tickUserList event fixed testing using this fn

Raihan9797 committed:
implement and test owner withdraw from main pool

Raihan9797 commented:
Replace console.log with emit events in testing
Implemented transfer to main pool when all users lose a habit challenge
Contract owner can withdraw() from the main_pool
Implemented testing for these questions

ambhinav reviewed (approved):
LGTM
